### PR TITLE
Half way done

### PR DESCRIPTION
**Pull request by karki011**
I have worked to tacoPraser and unit test.

### CODE REVIEW

File: LoggingKata.Test/ParserTest.cs [hunk 1]
```diff
@@ -4,16 +4,40 @@
...
+using LoggingKata;
 
 namespace LoggingKata.Test
 {
     [TestFixture]
     public class TacoParserTestFixture
     {
         [Test]
-        public void ShouldParseLine()
+        public void ShouldReturnNullForEmptyString()
```
Comment (codytwinton) [inline LoggingKata.Test/ParserTest.cs]: You should also write tests for:

- `null` string should return `null`
- `string` with lat and long, but no description returns valid `ITrackable`
- `string` without both lat and long should return `null`

File: LoggingKata/Program.cs [hunk 2]
```diff
@@ -16,6 +16,10 @@ class Program
 
         static void Main(string[] args)
         {
+            Logger.Info("Logging Started");
+            var path = Environment.CurrentDirectory + "\\Taco_Bell-US-AL-Alabama.csv";
+            Logger.Debug("Created a path to cvs file");
+
             if (args.Length == 0)
             {
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Missing implementation of parsing taco bells

File: LoggingKata/TacoParser.cs [hunk 3]
```diff
@@ -19,8 +21,21 @@ public TacoParser()
 
         public ITrackable Parse(string line)
         {
-            //DO not fail if one record parsing fails, return null
-            return null; //TODO Implement
+            var cells = line.Split(',');
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: Will fail for a `null` string

File: LoggingKata/TacoParser.cs [hunk 4]
```diff
@@ -19,8 +21,21 @@ public TacoParser()
...
+            if (cells.Length < 3)
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: Will fail for a string with lat and long, but no description

File: LoggingKata/TacoParser.cs [hunk 5]
```diff
@@ -19,8 +21,21 @@ public TacoParser()
...
+            {
+                Logger.Error("Length is less than 3, something went wrong");
+                return null;
+            }
+
+            Logger.Info("About to initialize object to get name and location of tacobell.");
+            var tacoBell = new TacoBell
+            {
+                Name = cells[2],
+                Location = new Point(double.Parse(cells[0]), double.Parse(cells[1]))
```
Comment (codytwinton) [inline LoggingKata/TacoParser.cs]: This will fail for non-valid numbers

File: LoggingKata.Test/ParserTest.cs (hunk 1 above)
Comment (karki011) [reply LoggingKata.Test/ParserTest.cs]: I think I have all checking working to my best of knowledge.

File: LoggingKata/TacoParser.cs (hunk 5 above)
Comment (karki011) [reply LoggingKata/TacoParser.cs]: i think i fixed it using try catch.

File: LoggingKata/TacoParser.cs (hunk 4 above)
Comment (karki011) [reply LoggingKata/TacoParser.cs]: i think i fixed it using try catch.

File: LoggingKata/TacoParser.cs (hunk 3 above)
Comment (karki011) [reply LoggingKata/TacoParser.cs]: i think i fixed it using try catch.

**Comment by codytwinton** [comment]
On your way.

**Comment by karki011** [comment]
I have refactored and added all the change.

File: LoggingKata.Test/ParserTest.cs [hunk 6]
```diff
@@ -13,7 +14,41 @@ public class TacoParserTestFixture
...
-            //TODO: Complete ShouldParseLine
+            //Arrange
+            var tacoParser = new TacoParser();
+            var testCoordination ="- 84.677017, 34.073638, Taco Bell Acwort... (Free trial * Add to Cart for a full POIinfo";
+
+            // Act 
+            var result = tacoParser.Parse(testCoordination);
+
+            // assert
+            Assert.Null(result);
```
Comment (codytwinton) [inline LoggingKata.Test/ParserTest.cs]: I believe this should be `IsNotNull`

File: LoggingKata.Test/ParserTest.cs [hunk 7]
```diff
@@ -13,7 +14,41 @@ public class TacoParserTestFixture
...
+        }
+
+        [Test]
+        public void ShouldNotParseLine()
+        {
+            //Arrange
+            var tacoParser = new TacoParser();
+
+            var empty = "";
+            var twoOutOfThreeCoordinates = "- 84.677017, 34.073638";
```
Comment (codytwinton) [inline LoggingKata.Test/ParserTest.cs]: I think this value should parse

File: LoggingKata/Program.cs [hunk 8]
```diff
@@ -5,32 +5,90 @@
...
         private static readonly ILog Logger =
             LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
 
         static void Main(string[] args)
         {
-            if (args.Length == 0)
+            Console.WriteLine("Program started");
+            var hostName = Dns.GetHostName();
+            Console.WriteLine(hostName);
+            var myIP = Dns.GetHostByName(hostName).AddressList[0].ToString();
```
Comment (codytwinton) [inline LoggingKata/Program.cs]: Nice.

File: LoggingKata/TacoParser.cs (hunk 3 above)
Comment (codytwinton) [reply LoggingKata/TacoParser.cs]: `var cells = line.Split(',');` should be placed after you `if (string.IsNullOrEmpty(line))` check.

**Pull request closed without merging**